### PR TITLE
First version of the package in ROS Noetic

### PR DESCRIPTION
**Pull request by FlorGrosso**
Adds a version of the package in ROS 1 Noetic, Python 3. It includes:

- `massrobotics_amr_sender` package which can be installed from source.
- sample scripts and rosbag to test the package in ROS 1.

TODOs:

- Add testing.
- Deploy the package to the ROS Buildfarm.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,15 +10,14 @@ standards, with a focus on AMRs (Autonomous Mobile Robots).
...
 The following packages are included in this repository:
 
+### Mass Robotics AMR Interop Sender for ROS1
 
-### Mass Robotics AMR Interop Sender for ROS2
-
-The [massrobotics_amr_sender_py](https://github.com/inorbit-ai/ros_amr_interop/tree/foxy-devel/massrobotics_amr_sender_py#readme)
-package provides a ROS2 node written in Python that takes input from a
-ROS2 system and publishes it to a [Mass Robotics Interop compliant
+The [massrobotics_amr_sender](https://github.com/inorbit-ai/ros_amr_interop/tree/foxy-devel/massrobotics_amr_sender_py#readme)
```
Comment (leandropineda) [inline README.md]:
```suggestion
The [massrobotics_amr_sender](https://github.com/inorbit-ai/ros_amr_interop/tree/noetic-devel/massrobotics_amr_sender_py#readme)
```

File: massrobotics_amr_sender/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+git clone --branch noetic-devel https://github.com/inorbit-ai/ros_amr_interop.git ./src
+# Install dependencies
+rosdep update && rosdep install --ignore-src --from-paths src/
+# Run the build
+catkin config --install
+catkin build
+```
+# Node configuration
+
+A configuration file must be provided to define how ROS1 messages are mapped to different AMR Interop Standard messages. A [sample_config.yaml](https://github.com/inorbit-ai/ros_amr_interop/blob/foxy-devel/massrobotics_amr_sender_py/sample_config.yaml) is provided for reference.
```
Comment (leandropineda) [inline massrobotics_amr_sender/README.md]:
```suggestion
A configuration file must be provided to define how ROS1 messages are mapped to different AMR Interop Standard messages. A [sample_config.yaml](https://github.com/inorbit-ai/ros_amr_interop/blob/noetic-devel/massrobotics_amr_sender_py/sample_config.yaml) is provided for reference.
```

File: massrobotics_amr_sender/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+# Running the sender node
+
+The node takes the MassRobotics AMR config file path as parameter. If not provided, it is assumed the file is on the current directory.
+
+```bash
+# Remember to source the ROS2 environment from the binary installation or your workspace overlay
```
Comment (leandropineda) [inline massrobotics_amr_sender/README.md]:
```suggestion
# Remember to source the ROS environment from the binary installation or your workspace overlay
```

File: massrobotics_amr_sender/launch/massrobotics_amr_sender.launch [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  POSSIBILITY OF SUCH DAMAGE.
+-->
+
+<launch>
```
Comment (leandropineda) [inline massrobotics_amr_sender/launch/massrobotics_amr_sender.launch:30]: Is the indentation below correct? If not, can you fix it?

File: massrobotics_amr_sender/sample/config.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # MassRobotics AMR Interoperability Standard sender configuration file
 # ====================================================================
 #
-# Parameters that are used to configure ROS2 node for connecting to MassRobotics compatible servers.
+# Parameters that are used to configure ROS1 node for connecting to MassRobotics compatible servers.
 # The `server` section expects a string with a WebSocket server URI, while the `mappings` section
 # contains a list of paramaters for configuring the ROS2 node. As per AMR Interop Standard,
```
Comment (leandropineda) [inline massrobotics_amr_sender/sample/config.yaml]:
```suggestion
# contains a list of paramaters for configuring the ROS node. As per AMR Interop Standard,
```

File: massrobotics_amr_sender/sample/massrobotics_amr_sender_rosbag_launch.launch [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  POSSIBILITY OF SUCH DAMAGE.
+-->
+
+<launch>
```
Comment (leandropineda) [inline massrobotics_amr_sender/sample/massrobotics_amr_sender_rosbag_launch.launch:30]: Please fix indentation

**Review by leandropineda — APPROVED** [review]
I did a first pass and it LGTM Flor. I left a couple of comments for nits only. Could you also look for other missing references to `foxy-devel` and `ROS2` on your branch before merging?

File: massrobotics_amr_sender/launch/massrobotics_amr_sender.launch (hunk 4 above)
Comment (FlorGrosso) [reply massrobotics_amr_sender/launch/massrobotics_amr_sender.launch:30]: Fixed!

File: massrobotics_amr_sender/sample/massrobotics_amr_sender_rosbag_launch.launch (hunk 6 above)
Comment (FlorGrosso) [reply massrobotics_amr_sender/sample/massrobotics_amr_sender_rosbag_launch.launch:30]: Fixed!

**Comment by FlorGrosso** [comment]
Done! Thanks Lean for checking that, I've updated all the references to ROS1.

File: massrobotics_amr_sender/README.md (hunk 2 above)
Comment (adamantivm) [reply massrobotics_amr_sender/README.md]: was this suggestion committed? Maybe it's just GitHub, but I see still the old path pointing to foxy-devel

**Review by adamantivm — APPROVED** [review]
I think this is awesome @FlorGrosso ! I would go ahead and merge. We can then test it and fix as we find problems.
If you want to be extra nice, you can create issues for the remaining items (tests and publishing the package)

**Comment by FlorGrosso** [comment]
Thanks. @adamantivm ! I've created #26 and #27.

**Pull request merged**